body/stretch_body/stepper.py

```diff
@@ -568,7 +571,10 @@ def menu_transaction(self,x,do_print=True):
             while self.transport.ser.inWaiting():
                 r=self.transport.ser.readline()
                 if do_print:
-                    print(r, end=' ')
+                    if type(r)==bytes:
```

Comment:
What does this do?

Reply:
The `self.transport.ser.readline()` method returns `byte` string type which does not neatly print the output to the terminal with the newlines when directly passed to the `print()` method. Therefore this step checks if the received lines from serial is in byte format, then it is decoded to `str` type and passed to the `print()` method to get neat outputs.
Here is an example of one of the outputs before and after decoding to string type.
Before:
```
b'Beginning calibration routine...\r\n' Hit enter after calibration done or Motor stops ticking...
.b'\r\n' b'0% ...................\r\n' b'10% ...................\r\n' b'20% ...................\r\n' b'30% ...................\r\n' b'40% ...................\r\n' b'50% ...................\r\n' b'60% ...................\r\n' b'70% ...................\r\n' b'80% ...................\r\n' b'90% ...................\r\n' b'Writing to flash 0x15E00 page size PSZ=3 \r\n' b' \r\n' b'Calibration complete!\r\n' b'The calibration table has been written to non-volatile Flash memory!\r\n' b' \r\n' b' \r\n' 
```
After:
```
Beginning calibration routine...
 0% ...................
 10% ...................
 20% ...................
 30% ...................
 40% ...................
 50% ...................
 60% ...................
 70% ...................
 80% ...................
 90% ...................
 Writing to flash 0x15E00 page size PSZ=3 

 Calibration complete!
 The calibration table has been written to non-volatile Flash memory!
```